### ISSUE TITLE
Should error on non-awaitable `__aenter__`/`__aexit__` methods

### ISSUE THREAD

**Issue by sharkdp** [astral-sh/ty#4124]
This should be an error, probably on the `async with` statement, or on the `__aenter__` definition. `__aenter__` should be async and return an awaitable:

```py
class C:
    def __aenter__(self) -> int:
        return 0

    async def __aexit__(self, exc_type, exc, tb) -> None:
        pass

async def f() -> None:
    async with C() as x:
        pass
```

Similarly, a bad (non-async) `__aexit__` is also not detected by ty.
